### PR TITLE
Only compute suppressedMethNames if really needed

### PR DESCRIPTION
**Pull request by forki**

### CODE REVIEW

File: src/fsharp/NameResolution.fs [hunk 1]
```diff
@@ -4208,17 +4212,27 @@ let ResolveCompletionsInTypeForItem (ncenv: NameResolver) nenv m ad statics ty (
             | Item.MethodGroup _ ->
                 // REVIEW: add a name filter here in the common cases?
                 let minfos = 
-                    let minfos =
-                        AllMethInfosOfTypeInScope ncenv.InfoReader nenv (None,ad) PreferOverrides m ty 
-                        |> List.filter minfoFilter
+                    let minfos = AllMethInfosOfTypeInScope ncenv.InfoReader nenv (None,ad) PreferOverrides m ty 
+                    if isNil minfos then [] else
+                     
+                    let suppressedMethNames = Zset.ofList String.order (pinfoMethNames @ einfoMethNames)
```
Comment (auduchinok) [inline src/fsharp/NameResolution.fs]: @forki Could you please tell what is the advantage of using Zset here instead of HashSet?

Comment (forki) [reply src/fsharp/NameResolution.fs]: I did not change that part. It was zset before.

Comment (auduchinok) [reply src/fsharp/NameResolution.fs]: Yes, I know. I'm just curious and thought you could know the rationale.

Comment (forki) [reply src/fsharp/NameResolution.fs]: I assume it's a relict from old ocaml compat times.

Comment (forki) [reply src/fsharp/NameResolution.fs]: I assume it's a relict from old ocaml compat times.

File: src/fsharp/NameResolution.fs [hunk 2]
```diff
@@ -4208,17 +4212,27 @@ let ResolveCompletionsInTypeForItem (ncenv: NameResolver) nenv m ad statics ty (
...
+                    let minfos = 
+                        minfos
+                        |> List.filter (minfoFilter suppressedMethNames)
         
+                    if isNil minfos then [] else
```
Comment (KevinRansom) [inline src/fsharp/NameResolution.fs]: @forki,

Whilst I applaud the desire to reduce the code change here,  I find the if (expr) then expr else idiom completely unreadable.

Can we indent the if then properly please.  Other than that, I think the changes look good.

Kevin

Comment (forki) [reply src/fsharp/NameResolution.fs]: We have that thing in multiple places.

**Comment by KevinRansom** [comment]
@forki, we do … it is unreadable in many places.

**Comment by cartermp** [comment]
I'd be 100% okay with going through the codebase and proper indenting scopes. It's thrown me off a few times before.

**Comment by KevinRansom** [comment]
@cartermp, @forki.

I am comfortable with not introducing new unindentented nested scopes, and moving away from them when we need to make changes in those files.  The risk of introducing bugs with a wholesale edit is more than I am comfortable with.

Kevin

**Comment by forki** [comment]
Seriously? It's such a good practice Imho.

Phillip Carter <notifications@github.com> schrieb am Mo., 24. Sep. 2018,
16:59:

**Comment by forki** [comment]
@KevinRansom I moved everthing to the right....

**Comment by forki** [comment]
Yeah. Sigh

Eugene Auduchinok <notifications@github.com> schrieb am Mo., 24. Sep. 2018,
19:22:

**Comment by KevinRansom** [comment]
@forki, thank you.

**Comment by forki** [comment]
ok it's still green

**Pull request merged**